pandas/_libs/tslibs/conversion.pyx

```diff
@@ -217,11 +217,35 @@ def ensure_timedelta64ns(arr: ndarray, copy: bool=True):
 
     Returns
     -------
-    result : ndarray with dtype timedelta64[ns]
-
+    ndarray[timedelta64[ns]]
     """
-    return arr.astype(TD64NS_DTYPE, copy=copy)
-    # TODO: check for overflows when going from a lower-resolution to nanos
+    assert arr.dtype.kind == "m", arr.dtype
+
+    if arr.dtype == TD64NS_DTYPE:
+        return arr.copy() if copy else arr
+
+    # Re-use the datetime64 machinery to do an overflow-safe `astype`
+    dtype = arr.dtype.str.replace("m8", "M8")
+    dummy = arr.view(dtype)
+    try:
+        dt64_result = ensure_datetime64ns(dummy, copy)
```

Comment:
I think it would be nicer to actually make ensure_datetime64 to accept both dt64 and td64 (and call it something else). but i suppose that might be some work and could be done in a follown.

Reply:
im optimistic that this will be fixed [upstream](https://github.com/numpy/numpy/issues/16352), at which point we wont need either of these functions